addons/web_editor/controllers/main.py

```diff
@@ -519,8 +520,9 @@ def shape(self, module, filename, **kwargs):
         """
         svg = None
         if module == 'illustration':
-            attachment = request.env['ir.attachment'].sudo().search([('url', '=like', request.httprequest.path), ('public', '=', True)], limit=1)
-            if not attachment:
+            attachment_id = unslug(filename.split('/')[-1])[1]
+            attachment = request.env['ir.attachment'].sudo().browse(attachment_id)
+            if not attachment or not attachment.public or attachment.type != 'binary':
```

Comment:
if not attachment.exists()

attachment =  model.browse(-1) => browseRecord(-1) == True

```python
In [9]: bool(self.browse(-1))                                                                                                                                                                               
Out[9]: True

In [10]: bool(self.browse(-1).exists())                                                                                                                                                                     
Out[10]: False

```